models/gif_cond_exp_multisynapse.cpp

```diff
@@ -0,0 +1,654 @@
+/*
+ *  gif_cond_exp_multisynapse.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "gif_cond_exp_multisynapse.h"
+
+#ifdef HAVE_GSL
+
+// C++ includes:
+#include <limits>
+#include <iomanip>
+#include <iostream>
+#include <cstdio>
+
+// Includes from libnestutil:
+#include "numerics.h"
+
+// Includes from nestkernel:
+#include "exceptions.h"
+#include "kernel_manager.h"
+#include "universal_data_logger_impl.h"
+
+// Includes from sli:
+#include "dict.h"
+#include "dictutils.h"
+#include "integerdatum.h"
+#include "doubledatum.h"
+
+#include "compose.hpp"
+#include "propagator_stability.h"
+#include "event.h"
+
+namespace nest
+{
+
+/* ----------------------------------------------------------------
+ * Recordables map
+ * ---------------------------------------------------------------- */
+
+RecordablesMap< gif_cond_exp_multisynapse >
+  gif_cond_exp_multisynapse::recordablesMap_;
+
+// Override the create() method with one call to RecordablesMap::insert_()
+// for each quantity to be recorded.
+template <>
+void
+RecordablesMap< gif_cond_exp_multisynapse >::create()
+{
+  // use standard names wherever you can for consistency!
+  insert_( names::V_m,
+    &gif_cond_exp_multisynapse::
+      get_y_elem_< gif_cond_exp_multisynapse::State_::V_M > );
+  insert_( names::E_sfa, &gif_cond_exp_multisynapse::get_E_sfa_ );
+  insert_( names::I_stc, &gif_cond_exp_multisynapse::get_I_stc_ );
```

Comment:
@heplesser can RecordablesMap deal with vectors? (i.e. can we create a ``get_gs`` function that would return all synaptic conductances?)
I think it would be best to give access to as much synaptic information as possible...
Otherwise we could at least give access to the total synaptic current...